File: Backtracking/78.subsets.php
<?php

class Solution
{
  private $nums;
  private $result = [];
  private $count;

  function subsets($nums)
  {
    $this->nums = $nums;
    $this->count = count($nums);
    $this->helper([]);
    return $this->result;
  }

  function helper(array $temp = null, int $counter = 0)
  {
    if ($counter == $this->count) {
      $this->result[] = $temp;
      return;
    }
    $element = $this->nums[$counter];
    ++$counter;
    $cp = $temp;
    $cp[] = $element;
    $this->helper($temp, $counter);
    $this->helper($cp, $counter);
  }
}

File: Backtracking/78.Subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        self.res, self.lenn = [], len(nums)

        def make_res(prev, i):
            if i == self.lenn:
                self.res.append(prev)
                return
            make_res(prev, i+1) 
            make_res(prev+[nums[i]], i+1)

        make_res([], 0)        
        return self.res
